pandas/core/arrays/sparse.py

```diff
@@ -900,8 +900,11 @@ def shift(self, periods=1):
         else:
             arr = self
 
+        if fill_value is None:
```

Comment:
is this not the same as the arrays/base.py impl? do we need this override at all?

Reply:
there is a small diff in those lines
`        subtype = np.result_type(np.nan, self.dtype.subtype)

        if subtype != self.dtype.subtype:
            # just coerce up front
            arr = self.astype(SparseDtype(subtype, self.fill_value))
        else:
            arr = self`
if I see correctly

Reply:
so you can do this in the Sparse version, then call the super method no (and pass in the dtype)

Reply:
yes, but that would require a new argument for the method in the base class, and therefore changing the signature of the method and all the versions of it in subclasses. If it is not a problem I would like to avoid adding more changes to PR if they are not required.
I can make those change as an independent PR if that works

Reply:
@jreback please let me know what you think